[FEATURE] Add a Core Mergers Section to the "Howtos" (#203)
TYPO3-Documentation/TYPO3CMS-Guide-ContributionWorkflow: Documentation/CoreMergers/Backport.rst

.. include:: ../Includes.txt

.. index::
   single: Gerrit; Backport a Change
   single: Backport a Change

.. _coreMergers-backport:
.. _lifeOfAPatch-backport:

=================
Backport a Change
=================

.. include:: /_includes/CoreMergers.rst.txt

> First wait until the review process was successfully completed for the most
> recent affected branch.

Reviewer: ```suggestion
Before you start, wait until the review process has been successfully completed for the most
recent affected branch.
```
Since the first thing under Use Gerrit is "First try to use..." we have two firsts. So let's make this one more of a pre-requisite.

   that you **don't alter the Change-ID** but remove every line (also empty ones)
   below it. After doing so hit the :guilabel:`Cherry Pick` button.

   .. figure:: /Images/External/Gerrit/CoreMergers/Backport2.png
      :class: with-shadow

      Adjust the commit message and click :guilabel:`Cherry Pick`

#. Review backport patch

   This creates a new Change on Gerrit that needs to be reviewed and
   merged once more.

   .. figure:: /Images/External/Gerrit/CoreMergers/Backport3.png
      :class: with-shadow

      A new change based on another branch is created. It has the status
      "Active" once more.

#. Merge the backport

   If the backport is a :ref:`Low brainer <Gerrit-low-brainers>` you can vote
   :guilabel:`+2` and continue to merge the backport. Otherwise it should be
   reviewed by at least one other Core Merger.

   .. figure::  /Images/External/Gerrit/CoreMergers/Backport4.png
      :class: with-shadow

      Merging the backport

Manual backport
===============

If the automatic backport fails, you need to manually cherry-pick the patch to the target branch. (e.g. cherry-pick the

> master patch onto your local (up to date) 10.4 branch) You will most likely need to adjust the code for the older branch.

Reviewer: ```suggestion
main patch onto your local (up to date) 10.4 branch) You will most likely need to adjust the code for the older branch.
```
Have we changed from 'master' to 'main' yet? I'm not sure....

Author: Nope, we are still on master
